common/include/pcl/console/print.h

```diff
@@ -46,14 +53,21 @@
 // PCL_INFO_STREAM("Info: this is a point: " << pcl::PointXYZ(1.0, 2.0, 3.0) << std::endl);
 // PCL_ERROR_STREAM("Error: an Eigen vector: " << std::endl << Eigen::Vector3f(1.0, 2.0, 3.0) << std::endl);
 // NOLINTBEGIN(bugprone-macro-parentheses)
-#define PCL_LOG_STREAM(LEVEL, STREAM, CSTR, ATTR, FG, ARGS) if(pcl::console::isVerbosityLevelEnabled(pcl::console::LEVEL)) { fflush(stdout); pcl::console::change_text_color(CSTR, pcl::console::ATTR, pcl::console::FG); STREAM << ARGS; pcl::console::reset_text_color(CSTR); }
+#define PCL_LOG_STREAM(LEVEL, ARGS) \
+  if(pcl::console::isVerbosityLevelEnabled(pcl::console::LEVEL)) \
+  { \
+    pcl::console::LogRecorder rec; \
+    rec << ARGS; \
+    pcl::console::LogRecord entry{pcl::console::LEVEL, rec.to_string()}; \
```

Comment:
Could there be a way to tell the stringstream of LogRecorder to directly operate on the message string of the LogRecord? Otherwise there are multiple allocations and copies here, I think.